src/core/executor.c

```diff
@@ -154,21 +163,44 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
+                case ARG_MANAGER_SOCKET: {
+                        int fd;
+
+                        fd = parse_fd(optarg);
+                        if (fd < 0)
+                                return log_error_errno(fd,
+                                                "Failed to parse manager socket \"%s\": %m",
+                                                optarg);
+
+                        (void) fd_cloexec(fd, true);
```

Comment:
please check for errors here, simply as a mechanism to validate that the fd actually has been passed